vue.config.js

```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  publicPath: '/',
+  publicPath: process.env.BASE_URL,
```

Comment:
Do you know that this works in all scenarios? I've never seen this used anywhere else. Do you have an example of it being set like this elsewhere? Or have a reference / SO answer that suggests this as a fix? If it is, I'm puzzled as to why this isn't mentioned in the VueJs online docs.

Reply:
Here there is an example: https://cli.vuejs.org/config/#publicpath
```
module.exports = {
  publicPath: process.env.NODE_ENV === 'production'
    ? '/production-sub-path/'
    : '/'
}
```
I just do the same things but in a simpler way.

Reply:
Also, there is the following article: https://cli.vuejs.org/guide/mode-and-env.html#using-env-variables-in-client-side-code. It says that using `BASE_URL`   without any code modifications should be enough to set a sub-path, but it doesn't work with Alerta and I don't know why. An issue in the VueJS repo would be a better approach to solve the problem but I cannot file it because I don't know how Alerta Web works.

Reply:
P. S. Now this hack works for me and if the `BASE_URL` variable is omitted all works as early. So nobody should meet problems.

Reply:
>It says that using BASE_URL without any code modifications should be enough to set a sub-path, but it doesn't work with Alerta and I don't know why.

As I said before, I suspect it's because of dynamic routing eg. `/alert/:id`, `/confirm/:token` and `/reset/:token`.  Does dynamic routing work when `BASE_URL` is set and when it isn't?

Reply:
Also, it all seems very circular as `publicPath` is used later to set `BASE_URL` ...

```
  env.BASE_URL = options.publicPath
```
https://github.com/vuejs/vue-cli/blob/e7af0d8fa374311693f14d1605e7ae442e43c3a6/packages/%40vue/cli-service/lib/util/resolveClientEnv.js#L10

Reply:
Another thing to consider is that I tried to make version backward compatible (as much as possible) with the previous Alerta API. So links sent in API v6 with hashes in them eg. `#/` needed to work with this web UI. To do that I had to add redirect code in the `index.html`. However, this could be breaking vuejs routing in some situations.

https://github.com/alerta/alerta-webui/blob/9e2d89aac2b4849a67c73f87e4b50ae58a6a57d3/public/index.html#L35-L43

Reply:
> Does dynamic routing work when `BASE_URL` is set and when it isn't?

There are some errors in my web console when I refresh a page with URL ` /alerta/alert/be04c4c5-81b6-4cae-9fa6-f4fff17a73a7` but I'm not shure that is related with `BASE_URL` because I see the same errors when I put `/alerta` directly to `vue.config.js`:
![Screenshot from 2019-04-18 17-17-22](https://user-images.githubusercontent.com/15159594/56367426-e64df680-61fd-11e9-8bad-0740d09bc0c8.png)

Reply:
That means it can't find your API server config. How is it working at all?

Reply:
And then Alerta works incorrectly:
![Screenshot from 2019-04-18 17-19-48](https://user-images.githubusercontent.com/15159594/56367600-404ebc00-61fe-11e9-99c7-751a0759dfea.png)

Reply:
> That means it can't find your API server config. How is it working at all?

It happens only after refreshing the page.

Reply:
Before refreshing all works fine with and without `BASE_URL` inside `vue.config.js`.

Reply:
I want to test it on https://beta.alerta.io/ but I cannot log in :(.

Reply:
![Screenshot from 2019-04-18 17-25-20](https://user-images.githubusercontent.com/15159594/56367894-fa462800-61fe-11e9-91d0-944f648b9c8a.png)

Reply:
Try now.

Reply:
Hm... Looks like it try to use incorrect config:
![Screenshot from 2019-04-18 17-34-24](https://user-images.githubusercontent.com/15159594/56368567-5e1d2080-6200-11e9-8686-a129ab27302a.png)
But you specify `endpoint` option at build time, right? So all works fine after reloading a page.

Reply:
As a result, I've built Alerta with `BASE_URL' and 'VUE_APP_ALERTA_ENDPOINT` and I don't see any errors - all works fine.

Reply:
And I don't know why env vars don't work as described in the docs. Maybe VueJS maintainers knows :).